test/dtypes.c

```diff
@@ -6436,16 +6436,17 @@ test__Float16(void)
     }
 
     if (is_little_endian) {
-        if (path->conv.u.lib_func != H5T__conv_noop) {
```

Comment:
In at least cygwin, `H5T__conv_noop` has a different address from `path->conv.u.lib_func`, so the two can't be compared directly. Instead, compare based on the conversion path name.